### PR TITLE
[DOCS] Removes the default size definition of thread pool types

### PR DESCRIPTION
**Pull request by szabosteve**
This PR deletes the default size value of thread pool types on the Thread pools page. Every specific thread pool has its own default size value, thread pool types don't have their own default sizes anymore.

Related issue: https://github.com/elastic/elasticsearch/issues/48771

### CODE REVIEW

**Comment by szabosteve** [comment]
@elasticmachine test this please

File: docs/reference/modules/threadpool.asciidoc [hunk 1]
```diff
@@ -20,16 +20,18 @@ There are several thread pools, but the important ones include:
...
 
 `get`::
     For get operations. Thread pool type is `fixed`
     with a size of `# of available processors`,
     queue_size of `1000`.
 
 `analyze`::
-    For analyze requests. Thread pool type is `fixed` with a size of 1, queue size of 16.
+    For analyze requests. Thread pool type is `fixed` with a size of 1, queue 
+    size of 16.
```
Comment (jrodewig) [inline docs/reference/modules/threadpool.asciidoc]:
```suggestion
    size of `16`.
```

File: docs/reference/modules/threadpool.asciidoc [hunk 2]
```diff
@@ -20,16 +20,18 @@ There are several thread pools, but the important ones include:
...
+    queue_size of `100`.
 
 `get`::
     For get operations. Thread pool type is `fixed`
     with a size of `# of available processors`,
     queue_size of `1000`.
 
 `analyze`::
-    For analyze requests. Thread pool type is `fixed` with a size of 1, queue size of 16.
+    For analyze requests. Thread pool type is `fixed` with a size of 1, queue
```
Comment (jrodewig) [inline docs/reference/modules/threadpool.asciidoc]:
```suggestion
    For analyze requests. Thread pool type is `fixed` with a size of `1`, queue 
```

Comment (jrodewig) [inline docs/reference/modules/threadpool.asciidoc]:
```suggestion
    For analyze requests. Thread pool type is `fixed` with a size of `1`, queue 
```

File: docs/reference/modules/threadpool.asciidoc [hunk 3]
```diff
@@ -49,8 +51,9 @@ There are several thread pools, but the important ones include:
     keep-alive of `5m` and a max of `min(10, (# of available processors)/2)`.
 
 `listener`::
-    Mainly for java client executing of action when listener threaded is set to true.
-    Thread pool type is `scaling` with a default max of `min(10, (# of available processors)/2)`.
+    Mainly for java client executing of action when listener threaded is set to 
+    true. Thread pool type is `scaling` with a default max of
```
Comment (jrodewig) [inline docs/reference/modules/threadpool.asciidoc]:
```suggestion
    `true`. Thread pool type is `scaling` with a default max of 
```

File: docs/reference/modules/threadpool.asciidoc [hunk 4]
```diff
@@ -176,15 +178,14 @@ processors: 2
...
-on a 16-core machine, set `processors` to 8). Note that this is an
-expert-level use-case and there's a lot more involved than just setting
-the `processors` setting as there are other considerations like changing
-the number of garbage collector threads, pinning processes to cores,
-etc.
+. If you are running multiple instances of {es} on the same host but want {es} 
+to size its thread pools as if it only has a fraction of the CPU, you should 
+override the `processors` setting to the desired fraction (for example, if 
+you're running two instances of {es} on a 16-core machine, set `processors` to 
+8). Note that this is an expert-level use case and there's a lot more involved
```
Comment (jrodewig) [inline docs/reference/modules/threadpool.asciidoc]: We can probably just remove the parentheses from `(for example...` and treat it as a regular sentence.

**Comment by jrodewig** [comment]
I left a few minor comments, but this looks good to me from a docs perspective.
I'll defer to @mayya-sharipova for final approval.

**Review by mayya-sharipova — APPROVED** [review]
@szabosteve thank you for the change. This looks good to me!

**Pull request merged**